compiler/rustc_hir_typeck/src/_match.rs

```diff
@@ -145,7 +152,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 other_arms.remove(0);
             }
 
-            prior_arm = Some((arm_block_id, arm_ty, arm_span));
+            if !arm_ty.is_never() {
+                prior_arm = Some((arm_block_id, arm_ty, arm_span));
```

Comment:
Can you add a short comment here explaining why we skip recording an arm if it diverges here?